modules/core/src/main/java/org/apache/ignite/internal/processors/platform/compute/PlatformCompute.java

```diff
@@ -100,6 +103,20 @@ public PlatformCompute(PlatformContext platformCtx, ClusterGroup grp, String pla
         computeForPlatform = (IgniteComputeImpl)grp.ignite().compute(platformGrp);
     }
 
+    /**
+     * Constructor.
+     *
+     * @param platformCtx Context.
+     * @param compute Compute.
+     * @param computeForPlatform Compute over platform-specific nodes.
+     */
+    private PlatformCompute(PlatformContext platformCtx, IgniteComputeImpl compute,
```

Comment:
What about a factory method `PlatformCompute withExecutor(string name)` instead of a different constructor?

Reply:
Sorry, not sure I understand, can you elaborate?